packages/jest-worker/src/workers/messageParent.ts

```diff
@@ -15,7 +16,21 @@ export default function messageParent(
   if (!isMainThread && parentPort != null) {
     parentPort.postMessage([PARENT_MESSAGE_CUSTOM, message]);
   } else if (typeof parentProcess.send === 'function') {
-    parentProcess.send([PARENT_MESSAGE_CUSTOM, message]);
+    try {
+      parentProcess.send([PARENT_MESSAGE_CUSTOM, message]);
+    } catch (e: any) {
+      if (/circular structure/.test(e?.message)) {
+        // We can safely send a message to the parent process again
+        // because previous sending was halted by "TypeError: Converting circular structure to JSON".
+        // But this time the message will be cleared from circular references.
+        parentProcess.send([
+          PARENT_MESSAGE_CUSTOM,
+          withoutCircularRefs(message),
+        ]);
+      } else {
+        throw e;
+      }
```

Comment:
```suggestion
    } catch (error) {
      if (error instanceof Error && /circular structure/.test(error.message)) {
        // We can safely send a message to the parent process again
        // because previous sending was halted by "TypeError: Converting circular structure to JSON".
        // But this time the message will be cleared from circular references.
        parentProcess.send([
          PARENT_MESSAGE_CUSTOM,
          withoutCircularRefs(message),
        ]);
      } else {
        throw error;
      }
```

Reply:
thanks! fixed